### PR TITLE
[Transformers] Support input images encoded with b64.encodebytes

### PR DESCRIPTION
**Pull request by MadhuM02**
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p> 

[Open in GitHub Codespaces](https://codespaces.new/MadhuM02/mlflow/pull/12087?quickstart=1)

#### Install mlflow from this PR

```
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/12087/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 12087
```

</p>
</details>

### Related Issues/PRs

 #xxx

### What changes are proposed in this pull request?

 With transformers 4.41.0, vision pipelines support input images encoded with base64.encodebytes api. previously only base64.b64encode is supported. Adding the same support for mlflow transformers image models. 

### How is this PR tested?

- [x] Existing unit/integration tests
- [x] New unit/integration tests
- [ ] Manual tests

### Does this PR require documentation update?

- [x] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [x] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

#### Should this PR be included in the next patch release?

`Yes` should be selected for bug fixes, documentation updates, and other small changes. `No` should be selected for new features and larger changes. If you're unsure about the release classification of this PR, leave this unchecked to let the maintainers decide.

<details>
<summary>What is a minor/patch release?</summary>

- Minor release: a release that increments the second part of the version number (e.g., 1.2.0 -> 1.3.0).
  Bug fixes, doc updates and new features usually go into minor releases.
- Patch release: a release that increments the third part of the version number (e.g., 1.2.0 -> 1.2.1).
  Bug fixes and doc updates usually go into patch releases.

</details>

- [ ] Yes (this PR will be cherry-picked and included in the next patch release)
- [x] No (this PR will be included in the next minor release)

### CODE REVIEW

**Comment by MadhuM02** [comment]
@BenWilson2 @serena-ruan can you please review this PR??

**Comment by MadhuM02** [comment]
@BenWilson2 , @serena-ruan can you please take a look whenever possible??

File: mlflow/transformers/__init__.py [hunk 1]
```diff
@@ -2552,7 +2552,10 @@ def is_base64_image(image):
         """Check whether input image is a base64 encoded"""
 
         try:
-            return base64.b64encode(base64.b64decode(image)).decode("utf-8") == image
+            return (
+                base64.b64encode(base64.b64decode(image)).decode("utf-8") == image
+                or base64.encodebytes(base64.b64decode(image)).decode("utf-8") == image
```
Comment (B-Step62) [inline mlflow/transformers/__init__.py:2557]: Can we keep `base64.b64decode(image)` as variable and reuse, so we don't need to decode twice?

**Comment by B-Step62** [comment]
@MadhuM02 Thank you for the contribution, and sorry for the super late response...!!

The change overall looks good to me, should be good to merge once https://github.com/mlflow/mlflow/pull/12087#discussion_r1620171110 is addressed.

File: mlflow/transformers/__init__.py (hunk 1 above)
Comment (MadhuM02) [reply mlflow/transformers/__init__.py]: makes sense, fixed.

**Comment by MadhuM02** [comment]
> @MadhuM02 Thank you so much for the contribution, and terribly sorry for the late response...!!
> 
> The change overall looks good to me, should be good to merge once [#12087 (comment)](https://github.com/mlflow/mlflow/pull/12087#discussion_r1620171110) is addressed.

thanks for reviewing, can you please take a relook?

**Comment by MadhuM02** [comment]
> @MadhuM02 Thank you so much for the contribution, and terribly sorry for the late response...!!
> 
> The change overall looks good to me, should be good to merge once [#12087 (comment)](https://github.com/mlflow/mlflow/pull/12087#discussion_r1620171110) is addressed.

@B-Step62 floating this up incase you forgot about this

**Review by B-Step62 — APPROVED** [review]
@MadhuM02 Terribly sorry for the late reply! Approved and merged. Thank you so much for the contribution!!

**Pull request merged**